### PR TITLE
Use prowjob name as buildId in Jenkins

### PR DESCRIPTION
**Pull request by kargakis**

### CODE REVIEW

File: prow/kube/prowjob.go [hunk 1]
```diff
@@ -89,8 +89,6 @@ type ProwJobStatus struct {
 	BuildID         string       `json:"build_id,omitempty"`
 	JenkinsQueueURL string       `json:"jenkins_queue_url,omitempty"`
 	JenkinsEnqueued bool         `json:"jenkins_enqueued,omitempty"`
-	// TODO(spxtr): Drop this in favor of just BuildID.
```
Comment (spxtr) [inline prow/kube/prowjob.go]: Hooray, a TODO assigned to me is disappearing!

**Comment by spxtr** [comment]
LGTM, please rebase.
/lgtm

**Comment by kargakis** [comment]
Done. I also dropped some dead code.

**Comment by kargakis** [comment]
Let's get the max_concurrency fix first in and then I will rebase this on top

/hold

**Comment by kargakis** [comment]
Didn't conflict

/hold cancel

**Pull request merged**